Structure_Engine/Query/HasReinforcement.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.Reinforcement;
+using BH.oM.Structure.SectionProperties;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns true if the Bar has a concrete section with reinforcement intent defined with at least one IBarReinforcement in it. False if not, or if the Bar, ConcreteSection or BarRebarIntent is null.")]
+        [Input("bar", "The Bar to check for reinforcement.")]
+        [Output("hasReinforcement", "Returns true if the Bar has reinforcement assigned, false otherwise.")]
+        public static bool HasReinforcement(this Bar bar)
+        {
+            if (bar == null)
+                return false;
```

Comment:
```suggestion
            if (bar.IsNull())
                return false;
```

Reply:
Actually, intentionally did not use those methods for this case. 

The reason for adding this method, was because the CreateFraming from a bar is crashing for concrete sections with default BarRebarIntent (null). What this method essentially want to check is if this is something you can extract rebars from (all non-null and some reinforcement actually defined) without throwing any errors that will just confuse things.

I could maybe agree using the IsNull for the bar itself, but everything in the ConcreteSection method is very intentionally not doing this (even for the section itself, as casting any other section to a ConcreteSection will result in a null, hence this can also be a valid null -> false case, which should not prompt any errors).

I think the IsNull methods are great for when things need to exit the method because of the null, and for those cases we ofc want to raise errors, but I do not see these methods as that type.

Does the above make sense?